compiler/rustc_passes/src/check_attr.rs

```diff
@@ -2381,6 +2382,18 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
             }
         }
     }
+
+    fn check_rustc_pub_transparent(&self, attr_span: Span, span: Span, attrs: &[Attribute]) {
```

Comment:
Can you check that this is applied to an ADT only?

Reply:
Well, I rely on `#[repr(transparent)]` being present, which, in turn, should probably guaranteed it's only applied to ADTs? Is it still worth it to check?

Reply:
Eh, I'd kinda rather we double check but I guess it doesn't matter 🤷

Reply:
I mean, it's an internal attribute so we don't need to do *any* validation; all of this is for the benefit. The less we can encode internal assumptions about how attributes are used, the better IMO. But it's kinda moot. Unless we get some `repr(transparent)` changes in the future, or something.